src/posix/platform/netif.cpp

```diff
@@ -505,6 +510,182 @@ static void UpdateLink(otInstance *aInstance)
     }
 }
 
+#if defined(__linux__)
+otError AddRtAttr(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, const void *aData, uint8_t aLen)
+{
+    uint8_t        len = RTA_LENGTH(aLen);
+    struct rtattr *rta;
+    otError        error = OT_ERROR_NONE;
+
+    VerifyOrExit(NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len) <= aMaxLen, error = OT_ERROR_INVALID_ARGS);
+
+    rta           = (struct rtattr *)((char *)(aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
+    rta->rta_type = aType;
+    rta->rta_len  = len;
+    if (aLen)
+    {
+        memcpy(RTA_DATA(rta), aData, aLen);
+    }
+    aHeader->nlmsg_len = NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len);
+exit:
+    return error;
+}
+
+int AddRtAttr32(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, uint32_t aData)
+{
+    return AddRtAttr(aHeader, aMaxLen, aType, &aData, sizeof(aData));
+}
+
+static void AddExternalRoute(const otIp6Prefix &aPrefix)
+{
+    struct
+    {
+        struct nlmsghdr header;
+        struct rtmsg    msg;
+        char            buf[128];
+    } req{};
+    unsigned char data[sizeof(in6_addr)];
+    char          addrBuf[OT_IP6_ADDRESS_STRING_SIZE];
+    unsigned int  ifIdx = otSysGetThreadNetifIndex();
+
+    VerifyOrExit(sNetlinkFd >= 0);
+    VerifyOrExit(sAddedExternalRoutesNum < kMaxExternalRoutesNum);
```

Comment:
In those `AddExternalRoute` and `DeleteExternalRoute` functions, we are silently ignore any failures, I believe we want to catch or log any errors.